### DISCUSSION TITLE
What is the intended result of creating a cluster via PUT /admin/v2/clusters/{cluster-name}?

### DISCUSSION THREAD

**Discussion by dameiss-tibco** [apache/pulsar#22602] [Q&A]
What is the intended result of creating a cluster via PUT /admin/v2/clusters/{cluster-name}? I can create a cluster named foobar (with the proper JSON body) with completely bogus values for brokerServiceUrl and serviceUrl, and the cluster is created. But it is obviously not useable since the URLs are invalid. It seems to create some kind of meta-data within the cluster since I can GET /admin/v2/clusters/foobar.

Am I missing something? Or is there an implicit assumption that the cluster definition being create reflects an actual cluster implemented somewhere? In which case it would seem the PUT call should fail if the URLs are invalid.

**Comment by visortelle** [comment]
@dameiss-tibco hi 👋

You'll need to use these cluster names in further operations. For example, the "create tenant" operation requires at least one allowed cluster. The "create namespace" operation also requires clusters when you have at least two clusters in your Pulsar instance.

> with completely bogus values for brokerServiceUrl and serviceUrl, and the cluster is created. But it is obviously not useable since the URLs are invalid

<img width="1438" alt="Screenshot 2024-04-26 at 8 36 57 PM" src="https://github.com/apache/pulsar/assets/9302460/662547af-26c7-4061-98dc-0ed85e096da6">

From the screenshot above you can see that there is some URL validation. At the same time, you're right that the cluster you register may be unavailable at the moment of its registration in another cluster.

In practice, it doesn't introduce any problems.
